address.go

```diff
@@ -681,3 +698,87 @@ func (a *AddressWitnessScriptHash) WitnessVersion() byte {
 func (a *AddressWitnessScriptHash) WitnessProgram() []byte {
 	return a.witnessProgram[:]
 }
+
+// AddressTaproot is an Address for a pay-to-taproot (P2TR) output. See BIP 341
+// for further details.
+type AddressTaproot struct {
+	hrp            string
```

Comment:
Extract this into a base `AddressSegwit` struct so we can save on the duplicaation here and above? This way it can also inherit all the shared methods used here.

Reply:
Makes sense, was able to remove a lot of duplicated code!